include/wabt/binary-reader.h

```diff
@@ -99,15 +106,20 @@ class BinaryReaderDelegate {
   /* Type section */
   virtual Result BeginTypeSection(Offset size) = 0;
   virtual Result OnTypeCount(Index count) = 0;
+  virtual Result OnRecursiveRange(Index start_index, Index type_count) = 0;
   virtual Result OnFuncType(Index index,
+                            GCTypeExtension* gc_ext,
```

Comment:
This structure is passed as a second argument, because it is a header, but it could go to the last argument since it is an extra (and optional) information. Which one you prefer?